std/process.d

```diff
@@ -806,6 +806,68 @@ private void setCLOEXEC(int fd, bool on) nothrow @nogc
     assert (wait(spawnProcess([prog.path, "foo", "bar"])) == 0);
 }
 
+version (Posix) @system unittest
+{
+    import core.sys.posix.fcntl : open, O_RDONLY;
+    import core.sys.posix.unistd : close;
+    import std.path : buildPath;
+    import std.algorithm.searching : canFind, findSplitBefore;
+    import std.functional : reverseArgs;
+    import std.array : split;
+    import std.conv : to;
+    static import std.file;
+
+    auto directory = uniqueTempPath();
+    std.file.mkdir(directory);
+    scope(exit) std.file.rmdirRecurse(directory);
+    auto path = buildPath(directory, "tmp");
+    std.file.write(path, null);
+    auto fd = open(path.tempCString, O_RDONLY);
+    scope(exit) close(fd);
+
+    // try /proc/<pid>/fd/ on linux
+    version (linux)
+    {
+        TestScript proc = "ls /proc/$$/fd";
+        auto procRes = execute(proc.path, null);
+        if (procRes.status == 0)
+        {
+            auto fdStr = fd.to!string;
+            assert (!procRes.output.split.canFind(fdStr));
```

Comment:
ah, yes, I should test that stdin, stdout and stderr haven't been closed.

btw, if it was just a problem with the script here and nothing was being found, then the next assert would fail anyway.